src/actions.ml

```diff
@@ -257,6 +257,55 @@ let send_doc_url ~bot_info ~github_repo_full_name job_info =
   | _ ->
       Lwt.return_unit
 
+let send_bench_url ~bot_info ~github_repo_full_name job_info (gh_owner, gh_repo)
+    ~external_id =
+  Lwt_io.print "Pushing status check for bench job.\n"
+  <&>
+  let url = f "https://gitlab.com/coq/coq/-/jobs/%d" job_info.build_id in
+  let state = job_info.build_status in
+  let context = "bench" in
+  url |> Uri.of_string |> Client.get
+  >>= fun (resp, _) ->
+  let status_code = resp |> Response.status |> Code.code_of_status in
+  if Int.equal 200 status_code then
+    let state, status, conclusion, description =
+      match state with
+      | "success" ->
+          ("success", COMPLETED, Some SUCCESS, "Bench completed successfully")
+      | "created" ->
+          ("pending", QUEUED, None, "Bench ready — Comment \"@coqbot bench\" to start bench")
+      | "running" ->
+          ("pending", IN_PROGRESS, None, "Bench in progress")
+      | "cancelled" | "canceled" ->
+          ("failure", COMPLETED, Some CANCELLED, "Bench cancelled")
+      | "failed" ->
+          ("failure", COMPLETED, Some NEUTRAL, "Bench completed with failures")
+      | _ ->
+          failwith (f "Error: send_bench_url received unknown state %s." state)
```

Comment:
You should not raise an anomaly for something as non-problematic like this, just print a warning as in the `job_action` function. And you should not even use the "Error" keyword. The fact that this keyword is pretty rare is useful so that we can retrieve the issues quickly.

Reply:
Right this is copied from job_action.

Reply:
I will remove it.

Reply:
Sorry not job_action but success or pending

Reply:
Yes, but `success_or_pending` is only called on pre-filtered states, so the failwith is only there to avoid getting an "incomplete pattern matching" warning (it is a true "impossible state").

Reply:
What can I put here instead? Should I raise and catch an exception instead?

Reply:
You should do a simple print, like the:

```ocaml
    | unknown_state ->
        Lwt_io.printf "Unknown job status: %s\n" unknown_state )
```

in the `job_action` function.

Reply:
The issue is that this branch is returning 4 values and not an Lwt.

Reply:
Right. I missed that. Then, my recommendation is to wrap the result in a result monad. That being said, you could also refactor the code of this function to return an Lwt instead.

Reply:
OK i've raised and handled an exception instead.